cloud/scope/powervs_cluster_test.go

```diff
@@ -153,3 +157,162 @@ func TestNewPowerVSClusterScope(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcileNetwork(t *testing.T) {
+	var (
+		mockPowerVS *mockP.MockPowerVS
+		mockCtrl    *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockPowerVS = mockP.NewMockPowerVS(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When GetDHCPServer returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.PowerVSClientFactory = func() (powervs.PowerVS, error) {
+			mockPowerVS.EXPECT().GetDHCPServer(gomock.Any()).Return(nil, fmt.Errorf("intentional error"))
+			return mockPowerVS, nil
+		}
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+
+		requeue, err := clusterScope.ReconcileNetwork()
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(requeue).To(BeFalse())
+	})
+	t.Run("When DHCPID is empty and GetNetworkByID returns error ", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.PowerVSClientFactory = func() (powervs.PowerVS, error) {
+			network := &models.Network{}
+			mockPowerVS.EXPECT().GetNetworkByID(gomock.Any()).Return(network, fmt.Errorf("intentional error"))
+			return mockPowerVS, nil
+		}
+		clusterScopeParams.IBMPowerVSCluster.Spec.DHCPServer.ID = nil
+		clusterScopeParams.IBMPowerVSCluster.Spec.Network = infrav1beta2.IBMPowerVSResourceReference{ID: ptr.To("networkID")}
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+
+		requeue, err := clusterScope.ReconcileNetwork()
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(requeue).To(BeFalse())
+	})
+	t.Run("When DHCPID is empty and networkID is not empty", func(t *testing.T) {
```

Comment:
would be good to verify if the [status is being set ](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/6ea02986330f33e150af9027c292eff68af53f5c/cloud/scope/powervs_cluster.go#L842)accordingly for this and the below cases?